tests/epics/adcore/test_drivers.py

```diff
@@ -90,10 +90,12 @@ async def wait_then_fail():
         await asyncio.sleep(0)
         set_mock_value(driver.detector_state, adcore.DetectorState.DISCONNECTED)
 
+    await wait_then_fail()
+
     acquiring = await adcore.start_acquiring_driver_and_ensure_status(
         driver, timeout=0.1
     )
-    await wait_then_fail()
-
-    with pytest.raises(ValueError):
+    with pytest.raises(
+        ValueError, match="Final detector state DetectorState.DISCONNECTED"
+    ):
```

Comment:
We should be starting setting up `wait_then_fail` before we schedule the task which should fail.